### ISSUE TITLE
Project documents to a flat table for reporting

### ISSUE THREAD

**Issue by jeremydmiller** [JasperFx/marten#396]
We've talked about this since we first started Marten, but never recorded the user story for it. Came up today in a discussion at work.

I think it might be how we pull off "map/reduce" w/ groupings in Marten.

**Comment by mdissel** [comment]
Can we make Marten so smart that it will create the necessary JS code to transform the fields on the server?

**Comment by jeremydmiller** [comment]
I don't think you'd need any JS code. It's really not that much different mechanically from the existing duplicated field functionality.

**Comment by mdissel** [comment]
Is this the functionality you have in your mind? (i have ;) )

Order.CompanyID is foreign key to Company. Mapping is based on propercase "ClassPropertyName"

```
public class Company
{
    public int ID { get; set; }
    public string Name { get; set; }
}

public class Order
{
    public int ID { get; set; }
    public int CompanyID { get; set; }
	public DateTime DateOrdered {get;set;}
}

Public class CompanyOrderDTO {
	public int CompanyID { get; set; }
	public string CompanyName { get; set; }
	public DateTime OrderDateOrdered {get;set;}
}

query.Query<CompanyOrderDTO, Company>()
                    .Include<Order>(x => x.CompanyID)
                    .Where(x => x.ID > 0)
```

**Comment by jeremydmiller** [comment]
@mdissel No, that's a different thing altogether. This is about projecting documents or event streams to a flat table for reporting purposes.

**Comment by mdissel** [comment]
Ok, but the CompanyOrderDTO is also a 'flat table'. 

Is something like i mean planned? the SELECT statement that marten should generate is something like

`SELECT c.data ->> 'name' as CompanyName, o.data ->> 'DateOrdered' as OrderDateOrdered  from mt_doc_company c inner join mt_doc_order o on c.id = o.companyid`
